File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.cpp
class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) {
//         unordered_map<string,int>rows;
//         int n=grid.size();
//         for(int i=0;i<n;i++){
//             string R;
//             for(int j=0;j<n;j++){
//                 R+=to_string(grid[i][j]);
//                 R+="+";
                
//             }
//             rows[R]++;
//         }
//         int cnt=0;
//         for(int j=0;j<n;j++){
//             string C;
//             for(int i=0;i<n;i++){
//                 C+=to_string(grid[i][j]);
//                 C+="+";
//             }
//             cnt+=rows[C];
//         }
//         return cnt;
        
        map<vector<int>,int>rows;
        int n=grid.size();
        for(int i=0;i<n;i++){
            rows[grid[i]]++;
        }
        int cnt=0;
        for(int j=0;j<n;j++){
            vector<int>temp;
            for(int i=0;i<n;i++){
                temp.push_back(grid[i][j]);
            }
            cnt+=rows[temp];
        }
        return cnt;
    }
};

File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.java
class Solution {
    public int equalPairs(int[][] grid) {
         int res = 0, n = grid.length;
    HashMap<ArrayList<Integer>, Integer> x = new HashMap<>();
	HashMap<ArrayList<Integer>, Integer> y = new HashMap<>();
	
    for(int i=0; i<n; i++){
        ArrayList<Integer> temp1=new ArrayList<>();
        ArrayList<Integer> temp2=new ArrayList<>();
        for(int j=0; j<n; j++){
            temp1.add(grid[i][j]);
            temp2.add(grid[j][i]);
        }
        
        x.put(temp1,x.getOrDefault(temp1, 0)+1);
        y.put(temp2, y.getOrDefault(temp2, 0)+1);
       
    }
 for(ArrayList i: x.keySet())
        if(y.containsKey(i))
            res += x.get(i)*y.get(i);
    return res;
    }
}